internal/operands/common-templates/reconcile.go

```diff
@@ -0,0 +1,106 @@
+package common_templates
+
+import (
+	templatev1 "github.com/openshift/api/template/v1"
+	core "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"kubevirt.io/ssp-operator/internal/common"
+	"path/filepath"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func AddWatchTypesToScheme(s *runtime.Scheme) error {
+	return templatev1.Install(s)
+}
+
+func WatchClusterTypes() []runtime.Object {
+	return []runtime.Object{
+		&rbac.ClusterRole{},
+		&rbac.ClusterRoleBinding{},
+		&core.Namespace{},
+	}
+}
+
+func WatchTypes() []runtime.Object {
+	return []runtime.Object{
+		&templatev1.Template{},
+	}
+}
+
+func Reconcile(request *common.Request) error {
+	for _, f := range []func(*common.Request) error{
+		reconcileGoldenImagesNS,
+		reconcileViewRole,
+		reconcileViewRoleBinding,
+		reconcileEditRole,
+		reconcileTemplates,
+	} {
+		if err := f(request); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func reconcileGoldenImagesNS(request *common.Request) error {
+	return common.CreateOrUpdateClusterResource(request, newGoldenImagesNS(goldenImagesNSname),
+		func(newRes, foundRes controllerutil.Object) {
+			newNS := newRes.(*core.Namespace)
+			foundNS := foundRes.(*core.Namespace)
+			foundNS.Spec = newNS.Spec
+		})
+}
+
+func reconcileViewRole(request *common.Request) error {
+	return common.CreateOrUpdateClusterResource(request, newViewRole(goldenImagesNSname),
+		func(newRes, foundRes controllerutil.Object) {
+			foundRes.(*rbac.ClusterRole).Rules = newRes.(*rbac.ClusterRole).Rules
+		})
+}
+
+func reconcileViewRoleBinding(request *common.Request) error {
+	return common.CreateOrUpdateClusterResource(request, newViewRoleBinding(goldenImagesNSname),
+		func(newRes, foundRes controllerutil.Object) {
+			newBinding := newRes.(*rbac.ClusterRoleBinding)
+			foundBinding := foundRes.(*rbac.ClusterRoleBinding)
+			foundBinding.RoleRef = newBinding.RoleRef
+			foundBinding.Subjects = newBinding.Subjects
+		})
+}
+
+func reconcileEditRole(request *common.Request) error {
+	return common.CreateOrUpdateClusterResource(request, newEditRole(goldenImagesNSname),
+		func(newRes, foundRes controllerutil.Object) {
+			newRole := newRes.(*rbac.ClusterRole)
+			foundRole := foundRes.(*rbac.ClusterRole)
+			foundRole.Rules = newRole.Rules
+		})
+}
+
+func reconcileTemplates(request *common.Request) error {
+	loadTemplates := func() {
+		filename := filepath.Join(bundleDir, "common-templates-"+version+".yaml")
+		readAndLoadTemplates(filename)
+	}
+	// Only load templates Once
+	loadTemplatesOnce.Do(loadTemplates)
```

Comment:
It seems confusing to use `sync.Once` variable from another file. Please either move the variables to this file or move this call to `resource.go`.

Reply:
Moved